x-pack/plugins/security_solution/server/lists_integration/endpoint/handlers/exceptions_pre_export_handler.ts

```diff
@@ -6,13 +6,29 @@
  */
 
 import { EndpointAppContextService } from '../../../endpoint/endpoint_app_context_services';
-import { ExtensionPoint } from '../../../../../lists/server';
+import { ExceptionsListPreExportServerExtension } from '../../../../../lists/server';
+import { TrustedAppValidator } from '../validators/trusted_app_validator';
 
+type ValidatedReturnType = ExceptionsListPreExportServerExtension['callback'];
 export const getExceptionsPreExportHandler = (
-  endpointAppContext: EndpointAppContextService
-): (ExtensionPoint & { type: 'exceptionsListPreExport' })['callback'] => {
-  return async function ({ data }) {
-    // Individual validators here
+  endpointAppContextService: EndpointAppContextService
+): ValidatedReturnType => {
+  return async function ({ data, context: { request, exceptionListClient } }) {
+    const { listId: maybeListId, id } = data;
+    let listId: string | null | undefined = maybeListId;
+
+    if (!listId && id) {
+      listId = (await exceptionListClient.getExceptionList(data))?.list_id ?? null;
+    }
+
+    if (!listId) {
+      return data;
+    }
+
+    // Validate Trusted Applications
+    if (TrustedAppValidator.isTrustedApp({ listId })) {
+      await new TrustedAppValidator(endpointAppContextService, request).validatePreExport();
```

Comment:
Here and in a few other places: suggest that you immediately `return data` after this check so that the additional artifact checks that will be added (soon) are not evaluated. there is no need to continue